datetime/datetime_test.go

```diff
@@ -0,0 +1,543 @@
+		}
+		if val, ok := tpl[dtIndex].(time.Time); !ok || !val.Equal(tm) {
+			fmt.Println("Tuple:   ", val)
+			fmt.Println("Expected:", tm)
+			t.Fatalf("Unexpected return value body (tuple %d field)", dtIndex)
+		}
+	}
+}
+
+func tupleInsertSelectDelete(t *testing.T, conn *Connection, tm time.Time) {
+	// Insert tuple with datetime.
+	_, err := conn.Insert(space, []interface{}{tm})
+	if err != nil {
+		t.Fatalf("Datetime insert failed: %s", err.Error())
+	}
+
+	// Select tuple with datetime.
+	var offset uint32 = 0
+	var limit uint32 = 1
+	resp, err := conn.Select(space, index, offset, limit, IterEq, []interface{}{tm})
+	if err != nil {
+		t.Fatalf("Datetime select failed: %s", err.Error())
+	}
+	if resp == nil {
+		t.Fatalf("Response is nil after Select")
+	}
+	assertDatetimeIsEqual(t, resp.Data, tm)
+
+	// Delete tuple with datetime.
+	resp, err = conn.Delete(space, index, []interface{}{tm})
+	if err != nil {
+		t.Fatalf("Datetime delete failed: %s", err.Error())
+	}
+	if resp == nil {
+		t.Fatalf("Response is nil after Delete")
+	}
+	assertDatetimeIsEqual(t, resp.Data, tm)
+}
+
+var datetimes = []string{
+	"2012-01-31T23:59:59.000000010Z",
+	"1970-01-01T00:00:00.000000010Z",
+	"2010-08-12T11:39:14Z",
+	"1984-03-24T18:04:05Z",
+	"2010-01-12T00:00:00Z",
+	"1970-01-01T00:00:00Z",
+	"1970-01-01T00:00:00.123456789Z",
+	"1970-01-01T00:00:00.12345678Z",
+	"1970-01-01T00:00:00.1234567Z",
+	"1970-01-01T00:00:00.123456Z",
+	"1970-01-01T00:00:00.12345Z",
+	"1970-01-01T00:00:00.1234Z",
+	"1970-01-01T00:00:00.123Z",
+	"1970-01-01T00:00:00.12Z",
+	"1970-01-01T00:00:00.1Z",
+	"1970-01-01T00:00:00.01Z",
+	"1970-01-01T00:00:00.001Z",
+	"1970-01-01T00:00:00.0001Z",
+	"1970-01-01T00:00:00.00001Z",
+	"1970-01-01T00:00:00.000001Z",
+	"1970-01-01T00:00:00.0000001Z",
+	"1970-01-01T00:00:00.00000001Z",
+	"1970-01-01T00:00:00.000000001Z",
+	"1970-01-01T00:00:00.000000009Z",
+	"1970-01-01T00:00:00.00000009Z",
+	"1970-01-01T00:00:00.0000009Z",
+	"1970-01-01T00:00:00.000009Z",
+	"1970-01-01T00:00:00.00009Z",
+	"1970-01-01T00:00:00.0009Z",
+	"1970-01-01T00:00:00.009Z",
+	"1970-01-01T00:00:00.09Z",
+	"1970-01-01T00:00:00.9Z",
+	"1970-01-01T00:00:00.99Z",
+	"1970-01-01T00:00:00.999Z",
+	"1970-01-01T00:00:00.9999Z",
+	"1970-01-01T00:00:00.99999Z",
+	"1970-01-01T00:00:00.999999Z",
+	"1970-01-01T00:00:00.9999999Z",
+	"1970-01-01T00:00:00.99999999Z",
+	"1970-01-01T00:00:00.999999999Z",
+	"1970-01-01T00:00:00.0Z",
+	"1970-01-01T00:00:00.00Z",
+	"1970-01-01T00:00:00.000Z",
+	"1970-01-01T00:00:00.0000Z",
+	"1970-01-01T00:00:00.00000Z",
+	"1970-01-01T00:00:00.000000Z",
+	"1970-01-01T00:00:00.0000000Z",
+	"1970-01-01T00:00:00.00000000Z",
+	"1970-01-01T00:00:00.000000000Z",
+	"1973-11-29T21:33:09Z",
+	"2013-10-28T17:51:56Z",
+	"9999-12-31T23:59:59Z",
+}
+
+func TestDatetimeInsertSelectDelete(t *testing.T) {
+	if isDatetimeSupported == false {
+		t.Skip("Skipping test for Tarantool without datetime support in msgpack")
+	}
```

Comment:
The code can be extracted into a separate function.

Reply:
What for it should be extracted to a separate test helper function?

Reply:
This will help you not to repeat the check. Something like that:
```go
func skipIfDatetimeUnsupported(t testing.TB) {
    t.Helper()
    if isDatetimeSupported == false {
        t.Skip("Skipping test for Tarantool without datetime support in msgpack")
    }
}
```

Reply:
Don't see any problem here.
Discussed verbally - it is a bad practice to disable/enable testcases. Feel free to ignore my opinion, you are the maintainer and final decision is yours.